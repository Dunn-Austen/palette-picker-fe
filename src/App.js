import React, {useState, useEffect} from 'react';
import './App.css';
import ColorContainer from './ColorContainer';
import SavePaletteForm from './SavePaletteForm';

const App = () => {

  let [palleteColor_1, setPaletteColor_1] = useState('');
  let [palleteColor_2, setPalleteColor_2] = useState('');
  let [palleteColor_3, setPalleteColor_3] = useState('');
  let [palleteColor_4, setPalleteColor_4] = useState('');
  let [palleteColor_5, setPalleteColor_5] = useState('');

  let [frozenStatus_1, setFrozenStatus_1] = useState(false);
  let [frozenStatus_2, setFrozenStatus_2] = useState(false);
  let [frozenStatus_3, setFrozenStatus_3] = useState(false);
  let [frozenStatus_4, setFrozenStatus_4] = useState(false);
  let [frozenStatus_5, setFrozenStatus_5] = useState(false);

  const generateRandomColors = () => {
    if (!frozenStatus_1) setPaletteColor_1('#'+Math.floor(Math.random()*16777215).toString(16));
    if (!frozenStatus_2) setPaletteColor_2('#'+Math.floor(Math.random()*16777215).toString(16));
    if (!frozenStatus_3) setPaletteColor_3('#'+Math.floor(Math.random()*16777215).toString(16));
    if (!frozenStatus_4) setPaletteColor_4('#'+Math.floor(Math.random()*16777215).toString(16));
    if (!frozenStatus_5) setPaletteColor_5('#'+Math.floor(Math.random()*16777215).toString(16));
  }

  useEffect(() => generateRandomColors(), []);



  return (
    <main className='app'>
      <section className='colors-section'>
        <ColorContainer
          paletteColors={paletteColor_1, paletteColor_2, paletteColor_3, paletteColor_4, paletteColor_5}
          frozenStatuses={frozenStatus_1, frozenStatus_2, frozenStatus_3, frozenStatus_4, frozenStatus_5}
        />
      </section>
    </main>
  );
}

export default App;
