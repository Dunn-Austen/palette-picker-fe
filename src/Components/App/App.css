.app {
  height: 100vh;
  text-align: center;
}

.colors-section {
  height: 40vh;
  background-color: white;
}

.generate-palette {
  background-color: beige;
  border-radius: 20px;
  font-size: 25px;
  height: 35px;
  margin-top: 5px;
  width: 300px;
}

.generate-palette:hover {
  background-color: yellow;
  border: 2px solid black;
}

.forms-section {
  align-items: center;
  background-color: beige;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 25vh;
}
