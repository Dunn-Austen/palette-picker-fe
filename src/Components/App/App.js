import React, {useState, useEffect} from 'react';
import './App.css';
import ColorContainer from '../ColorContainer/ColorContainer';
import SavePaletteForm from '../SavePaletteForm/SavePaletteForm';

const App = () => {

  let [paletteColor_1, setPaletteColor_1] = useState('#000000');
  let [paletteColor_2, setPaletteColor_2] = useState('#bbbbbb');
  let [paletteColor_3, setPaletteColor_3] = useState('#222222');
  let [paletteColor_4, setPaletteColor_4] = useState('#dddddd');
  let [paletteColor_5, setPaletteColor_5] = useState('#444444');

  let [isSaved_1, setIsSaved_1] = useState(false);
  let [isSaved_2, setIsSaved_2] = useState(false);
  let [isSaved_3, setIsSaved_3] = useState(false);
  let [isSaved_4, setIsSaved_4] = useState(false);
  let [isSaved_5, setIsSaved_5] = useState(false);

  let [palette, setPalette] = useState('');
  let [projects, setProjects] = useState([]);

  const generateRandomColors = () => {
    if (!isSaved_1) {
      setPaletteColor_1('#'+Math.floor(Math.random()*16777215).toString(16))
    };
    if (!isSaved_2) {
      setPaletteColor_2('#'+Math.floor(Math.random()*16777215).toString(16))
    };
    if (!isSaved_3) {
      setPaletteColor_3('#'+Math.floor(Math.random()*16777215).toString(16))
    };
    if (!isSaved_4) {
      setPaletteColor_4('#'+Math.floor(Math.random()*16777215).toString(16))
    };
    if (!isSaved_5) {
      setPaletteColor_5('#'+Math.floor(Math.random()*16777215).toString(16))
    };
  }

  useEffect(() => generateRandomColors(), []);

  return (
    <main className='app'>
      <section className='colors-section'>
        <ColorContainer
          paletteColors={{paletteColor_1, paletteColor_2, paletteColor_3, paletteColor_4, paletteColor_5}}
          isSavedStatuses={{isSaved_1, isSaved_2, isSaved_3, isSaved_4, isSaved_5}}
          setIsSavedFunctions={{setIsSaved_1, setIsSaved_2, setIsSaved_3, setIsSaved_4, setIsSaved_5}}
        />
        <button className='generate-palette' onClick={() => generateRandomColors()}>
          Generate New Colors
        </button>
        <SavePaletteForm
          setPalette={setPalette}
        />
      </section>
    </main>
  );
}

export default App;
