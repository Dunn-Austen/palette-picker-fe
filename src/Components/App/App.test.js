import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

descrive('apiCalls', () => {
   
  describe('fetchAllProjects', () => {
    let mockResponse

    beforeEach(() => {
      mockResponse = {
        projects: [
          {
            id: 1, 
            title: "Kitchen"
          }, 
          {
            id: 2, 
            title: "Bedroom"
          }
        ]
      };
      window.fetch = jest.fn().mockImplementaion(() => {
        return Promise.resolve({
          ok: true,
          json: () => Promise.resolve(mockResopnse)
        });
      });
    });

    it('should call fetch with the correct URL', () => {
      fetchAllProjects();

      expect(window.fetch).toHaveBeenCalledWith('https://palette-picker-1908.herokuapp.com/api/v1/projects');
    })
  })
})
