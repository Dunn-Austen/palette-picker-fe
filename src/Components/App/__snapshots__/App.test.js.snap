// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="app"
>
  <section
    className="colors-section"
  >
    <ColorContainer
      isSavedStatuses={
        Object {
          "isSaved_1": false,
          "isSaved_2": false,
          "isSaved_3": false,
          "isSaved_4": false,
          "isSaved_5": false,
        }
      }
      paletteColors={
        Object {
          "paletteColor_1": "#000000",
          "paletteColor_2": "#bbbbbb",
          "paletteColor_3": "#222222",
          "paletteColor_4": "#dddddd",
          "paletteColor_5": "#444444",
        }
      }
      setIsSavedFunctions={
        Object {
          "setIsSaved_1": [Function],
          "setIsSaved_2": [Function],
          "setIsSaved_3": [Function],
          "setIsSaved_4": [Function],
          "setIsSaved_5": [Function],
        }
      }
    />
    <button
      className="generate-palette"
      onClick={[Function]}
    >
      Generate New Colors
    </button>
  </section>
  <section
    className="forms-section"
  >
    <SaveProjectForm
      addProject={[Function]}
      projects={Array []}
    />
    <SavePaletteForm
      paletteColors={
        Object {
          "paletteColor_1": "#000000",
          "paletteColor_2": "#bbbbbb",
          "paletteColor_3": "#222222",
          "paletteColor_4": "#dddddd",
          "paletteColor_5": "#444444",
        }
      }
      projects={Array []}
      updateProjects={[Function]}
    />
  </section>
</main>
`;
